.github/workflows/check_pr.yml

```diff
@@ -0,0 +1,35 @@
+name: Check branch conformity
+on:
+  pull_request:
+      types: [opened, synchronize, reopened]
+
+jobs:
+  prevent-fixup-commits:
+      runs-on: ubuntu-latest
+      env:
+        target: debian-bookworm
+        distro: debian
+        version: bookworm
+      steps:
+        - name: Checkout
+          uses: actions/checkout@v4
+        - name: prevent fixup commits
+          run: |
+            git fetch origin
+            git status
+            git log --pretty=format:%s origin/main..HEAD | grep -ie '^fixup\|^wip' && exit 1 || true
+
+  check-changelog:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: ensure CHANGELOG.md is populated
+        run: |
+          CHANGELOG_MODIFIED=$(git diff --name-only ${GITHUB_SHA} | grep -c "CHANGELOG.md")
+          if [ $CHANGELOG_MODIFIED -eq 0 ]; then
+            echo "::error::No CHANGELOG modifications found in this pull request. Please update the CHANGELOG."
```

Comment:
Suggestion: Since not every PR needs to update the changelog (but most PRs do), I think we could:

1. Replace `::error::` with `::warning::` (see https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/workflow-commands-for-github-actions#setting-a-warning-message)
2. Point to our `CONTRIBUTING.md` guide, in a section where we explain to contributors how to do so.
    * (not sure how easy it is, let's not do it if it's hard) We could even propose a message, by taking the PR title (is it available?), the contributor name, and the PR number.

Reply:
Thanks for the proposed changes. I've updated the code to use a warning rather than an error, it's a good idea :+1: 

About updating the `CONTRIBUTING.md` guide, I haven't found anything like that. Maybe I'm not looking at the right place :thinking: ?

If we don't already have somewhere to put this information, I propose to just add a warning and enhance / create a contributing guide as part of #903.

Reply:
My mistake, sorry! Yeap, what you're saying makes sense.